src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1290,16 +1287,14 @@ private static BoundNode Rewrite(ImmutableDictionary<BoundExpression, (Nullabili
             return rewrittenNode;
         }
 
+        private static bool CanSkipAnalysis(CSharpCompilation compilation)
```

Comment:
>CanSkipAnalysis [](start = 28, length = 15)

I think the name of this method is somewhat confusing given that sometimes we don't skip analysis when this method returns true. Consider renaming to "NeedResultFromAnalysis"/"NeedResultOfAnalysis" and reversing the logic appropriately.